apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/ConfigsExportController.java

```diff
@@ -0,0 +1,115 @@
+import com.ctrip.framework.apollo.common.exception.BadRequestException;
+import com.ctrip.framework.apollo.common.exception.ServiceException;
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.core.enums.ConfigFileFormat;
+import com.ctrip.framework.apollo.core.enums.Env;
+import com.ctrip.framework.apollo.portal.entity.bo.NamespaceBO;
+import com.ctrip.framework.apollo.portal.entity.model.NamespaceTextModel;
+import com.ctrip.framework.apollo.portal.service.ItemService;
+import com.ctrip.framework.apollo.portal.service.NamespaceService;
+import com.ctrip.framework.apollo.portal.util.ConfigToFileUtils;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * jian.tan
+ */
+@RestController
+public class ConfigsExportController {
+
+  @Autowired
+  private ItemService configService;
+
+  @Autowired
+  private NamespaceService namespaceService;
+
+  @RequestMapping(value = "/apps/{appId}/envs/{env}/clusters/{clusterName}/namespaces/{namespaceName}/items/import", method = RequestMethod.POST)
+  public void importConfigFile(@PathVariable String appId, @PathVariable String env,
+      @PathVariable String clusterName, @PathVariable String namespaceName,
+      @RequestParam Integer namespaceId,
+      @RequestParam("file") MultipartFile file) {
+    if (file.isEmpty()) {
+      throw new BadRequestException("The file is empty.");
+    }
+
+    NamespaceDTO namespaceDTO = namespaceService
+        .loadNamespaceBaseInfo(appId, Env.fromString(env), clusterName, namespaceName);
+
+    if (Objects.isNull(namespaceDTO)) {
+      throw new BadRequestException(String.format("Namespace: {} not exist.", namespaceName));
+    }
+
+    NamespaceTextModel model = new NamespaceTextModel();
+    List<String> fileNameSplit = Splitter.on(".").splitToList(file.getOriginalFilename());
+    if (fileNameSplit.size() <= 1) {
+      throw new BadRequestException("The file format is invalid.");
+    }
+
+    String format = fileNameSplit.get(fileNameSplit.size() - 1);
+    model.setFormat(format);
+    model.setAppId(appId);
+    model.setEnv(env);
+    model.setClusterName(clusterName);
+    model.setNamespaceName(namespaceName);
+    model.setNamespaceId(namespaceId);
+    String configText;
+    try {
+      configText = ConfigToFileUtils.fileToString(file.getInputStream());
```

Comment:
According to the java doc of `org.springframework.web.multipart.MultipartFile#getInputStream`: `The user is responsible for closing the returned stream.`, I think we should close it in the finally block

Reply:
yep, We cloud also use `try-with-resources` to handle it.

Reply:
either will do, :-)